src/editor/serialize.ts

```diff
@@ -38,7 +40,25 @@ export function mdSerialize(model: EditorModel) {
 }
 
 export function htmlSerializeIfNeeded(model: EditorModel, {forceHTML = false} = {}) {
-    const md = mdSerialize(model);
+    let md = mdSerialize(model);
+
+    if (SdkConfig.get()['latex_maths']) {
+        const displayPattern = (SdkConfig.get()['latex_maths_delims'] || {})['display_pattern'] ||
+            "\\$\\$\\$(([^$]|\\\\\\$)*)\\$\\$\\$";
+        const inlinePattern = (SdkConfig.get()['latex_maths_delims'] || {})['inline_pattern'] ||
+            "\\$\\$(([^$]|\\\\\\$)*)\\$\\$";
+
+        md = md.replace(RegExp(displayPattern, "gm"), function(m, p1) {
+            const p1e = AllHtmlEntities.encode(p1);
+            return `<div data-mx-maths="${p1e}"><code>${p1e}</code></div>`;
+        });
+
+        md = md.replace(RegExp(inlinePattern, "gm"), function(m, p1) {
```

Comment:
Since the maths processing is done before the markdown processing, there is the potential for some odd interactions.  If the LaTeX happens to have valid markdown, then the fallback will have the markdown applied.  For example, if someone types `$$ *x* $$`, then it will produce `<span data-mx-maths=\" *x* \"><code> <em>x</em> </code></span>` (note the `<em>x</em>` in the fallback).

It seems like this type of thing will be rare, and fortunately neither `$$ x*y*z $$` nor `$$ x * y * z $$` seem to trigger this.  But it would be nice if we could avoid this if possible.  It may be necessary to fiddle with the `toHtml` function in `src/Markdown.js`, and maybe see if you can coerce commonmark to output the literal contents of `<code>` blocks inside of data-mx-maths tags.

Reply:
after trying all sorts of stuff to get commonmark to ignore bits of text, i found the much simpler solution was to just add the `<code>` block with tex in after markdown processing (cc713af). Now the only change needed in Markdown.js is allowing the math tags through.

Reply:
That seems like a nice solution.